### ISSUE TITLE
[Question]: Running Playwright on .NET Azure Function (v4) on Windows

### ISSUE THREAD

**Issue by hasen011** [microsoft/playwright-dotnet#2348]
### Your question

Hi,

I am trying to run a .NET app with Playwright as an Azure Function (v4) hosted on Windows. I believe all the relevant threads and hasn't been successfully with running it.

I am using the API to install chromium (and other browsers)
`Microsoft.Playwright.Program.Main(new[] { "install" });`

I was able to get that to work after specifying a browser path (I tried other paths with no luck)
`PLAYWRIGHT_BROWSERS_PATH = "D:\home"`

However, after installing the browser, I ran into an error about missing dependencies
`FailureException: System.AggregateException: One or more errors occurred. (Host system is missing dependencies!Full list of missing libraries:chrome_elf.dll)`

After that, I also tried running 
`Microsoft.Playwright.Program.Main(new[] { "install-deps" });`

but that fails. It seems like that the 'install-deps' command is executing a powershell script 'install_media_pack.ps1' which executes the following
```
$osInfo = Get-WmiObject -Class Win32_OperatingSystem
# check if running on Windows Server
if ($osInfo.ProductType -eq 3) {
  Install-WindowsFeature Server-Media-Foundation
}
```
which seems like cannot be executed through the Azure Function.
`Get-WmiObject : No such interface supported`

Now, my question is - does anyone know if I can make this work or if I am headed towards a dead end?

Thank you in advance!

**Comment by mxschmitt** [comment]
The Azure Functions Windows environment is most likely too minimal to run a whole Browser in it. We recommend using Linux instead, which should work or running it with Azure Container Apps, where you can specify the [official Docker image](https://playwright.dev/docs/docker).

**Comment by MrSimonC** [comment]
@hasen011 Hey. So I'm currently running Azure Functions (linux isolated) which work great like this. I did all this about 9 months ago - .net 6.0. See my template-repo for details: https://github.com/MrSimonC/Playwright-In-Azure-Function-Template 

However, trying to reuse my template (yesterday) I'm just currently working through an error at the moment where `PLAYWRIGHT_BROWSERS_PATH` (see [docs](https://playwright.dev/dotnet/docs/browsers#managing-browser-binaries)) doesn't seem to be obeyed by playwright. I'll confirm back on here if I get it working - and will update my template repo. 

Checking the current `main` branch of playwright.net it seems the only place it appears is in the tests, or dockerfile - which seems very strange...

**Comment by hasen011** [comment]
@MrSimonC Thank you for your response. I ended up running it as an Azure Function as a docker image (linux) and that works fine. I am using the .NET api `Microsoft.Playwright.Program.Main(new[] { "install-deps" });` to install browsers and dependencies with a (Visual studio) auto generated docker file.

Never really figured it out to install everything through the docker file but don't mind this approach either. It just installs everything on the first run and then it's good to go.

**Comment by mxschmitt** [comment]
Since https://github.com/microsoft/playwright-dotnet/issues/2354 is merged and available in 1.28 I'm closing it.
